packages/server/src/workers/scheduledtiming.ts

```diff
@@ -0,0 +1,290 @@
+import { MedplumRedisConfig } from '../config';
+import { executeBot } from '../fhir/operations/execute';
+import { systemRepo } from '../fhir/repo';
+import { logger } from '../logger';
+import { AuditEventOutcome } from '../util/auditevent';
+import { isValidCron } from 'cron-validator';
+
+const daysOfWeekConversion = { sun: 0, mon: 1, tue: 2, wed: 3, thu: 4, fri: 5, sat: 6 };
+
+/*
+ * The ScheduleTiming worker inspects resources takes a bot,
+ * if it has the scheduledTiming property, will add it as a repeatable
+ * Cron job
+ */
+// Repeatable is based on BullMQ docs https://docs.bullmq.io/guide/jobs/repeatable
+interface Repeatable {
+  repeat: {
+    pattern?: string;
+    every?: number;
+    limit?: number;
+  };
+  jobId?: string;
+}
+
+export interface ScheduledTimingJobData {
+  readonly resourceType: string;
+  readonly id: string;
+}
+
+const queueName = 'ScheduledTimingQueue';
+const jobName = 'ScheduledTimingJobData';
+
+let queue: Queue<ScheduledTimingJobData> | undefined = undefined;
+let worker: Worker<ScheduledTimingJobData> | undefined = undefined;
+
+/**
+ * Initializes the scheduled timing worker.
+ * Sets up the BullMQ job queue.
+ * Sets up the BullMQ worker.
+ */
+export function initScheduledTimingWorker(config: MedplumRedisConfig): void {
+  const defaultOptions: QueueBaseOptions = {
+    connection: config,
+  };
+
+  queue = new Queue<ScheduledTimingJobData>(queueName, {
+    ...defaultOptions,
+    defaultJobOptions: {
+      attempts: 3,
+      backoff: {
+        type: 'exponential',
+        delay: 1000,
+      },
+    },
+  });
+  worker = new Worker<ScheduledTimingJobData>(queueName, execBot, defaultOptions);
+  worker.on('completed', (job) => logger.info(`Completed job ${job.id} successfully`));
+  worker.on('failed', (job, err) => logger.info(`Failed job ${job?.id} with ${err}`));
+}
+
+/**
+ * Shuts down the ScheduledTiming worker.
+ * Closes the BullMQ job queue.
+ * Clsoes the BullMQ worker.
+ */
+export async function closeScheduledTimingWorker(): Promise<void> {
+  if (queue) {
+    await queue.close();
+    queue = undefined;
+  }
+
+  if (worker) {
+    await worker.close();
+    worker = undefined;
+  }
+}
+
+/**
+ * Returns the ScheduledTiming queue instance.
+ * This is used by the unit tests.
+ * @returns The ScheduledTiming queue (if available).
+ */
+export function getScheduledTimingQueue(): Queue<ScheduledTimingJobData> | undefined {
+  return queue;
+}
+
+/**
+ * @param resource The resource that was created or updated.
+ */
+export async function addScheduledTimingJobs(resource: Resource): Promise<void> {
+  if (resource.resourceType !== 'Bot') {
+    // For now we have only the bot to execute on a timed job
+    return;
+  }
+  let cron;
+  // Validate the cron format
+  if (resource.cronTiming) {
+    cron = convertTimingToCron(resource.cronTiming);
+    if (!cron) {
+      logger.debug('cronTiming had the wrong format for a timed cron job');
```

Comment:
This is a good design question -- how can we provide feedback about invalid cron expressions.

I think this is an ok starting point, because we're still early in the cron development.

In the future, we may want to explore leveraging capabilities in `ElementDefinition.constraint` to make it a write-time check: https://hl7.org/fhir/elementdefinition-definitions.html#ElementDefinition.constraint

Reply:
Just filed a ticket to track this: [#1751](https://github.com/medplum/medplum/issues/1751)